universal_transfer_operator/src/universal_transfer_operator/data_providers/base.py

```diff
@@ -14,6 +17,13 @@
 DatasetType = TypeVar("DatasetType", File, Table)
 
 
+@attr.define
+class DataStream:
+    remote_obj_buffer: io.IOBase
+    actual_filename: Path
+    actual_file: File
```

Comment:
What are `actual_filename` and `actual_file`? Are they the local representation of a remote file?

Reply:
`actual_filename` is the name of source dataset which is to be transferred to destination. `actual_file` is local representation of remote file

Reply:
@sunank200 I think we can improve the naming in a separate PR, WDYT?

Reply:
Sure @tatiana I can create an issue for this in new repo.